utils/build/docker/golang/app/chi/main.go

```diff
@@ -285,9 +287,23 @@ func main() {
 		w.WriteHeader(http.StatusOK)
 	})
 
+	srv := &http.Server{
+		Addr:    ":7777",
+		Handler: mux,
+	}
+
+	c := make(chan os.Signal, 1)
+	signal.Notify(c, syscall.SIGTERM)
+	go func() {
+		for range c {
+			srv.Shutdown(context.Background())
+			return
+		}
+	}()
+
 	common.InitDatadog()
 	go grpc.ListenAndServe()
-	http.ListenAndServe(":7777", mux)
+	srv.ListenAndServe()
```

Comment:
it is better to do it the other way around - run `ListenAndServe` in a goroutine and block with the signal channel in the main goroutine + call `Shutdown`. The reason is the documentation of `Shutdown` says:

> When Shutdown is called, [Serve](https://pkg.go.dev/net/http#Serve), [ListenAndServe](https://pkg.go.dev/net/http#ListenAndServe), and [ListenAndServeTLS](https://pkg.go.dev/net/http#ListenAndServeTLS) immediately return [ErrServerClosed](https://pkg.go.dev/net/http#ErrServerClosed). Make sure the program doesn't exit and waits instead for Shutdown to return.

(more detailed explanation [here](https://dev.to/mokiat/proper-http-shutdown-in-go-3fji))